### PR TITLE
Add TS1 mechanism configuration and initial condtions to the examples

### PR DESCRIPTION
**Pull request by carl-drews**
Added the following files to examples:
 Directory of C:\2024\MusicBox\music-box-interactive-api\interactive\api\static\examples\TS1

08/01/2024  05:19 PM    <DIR>          .
08/01/2024  05:19 PM    <DIR>          ..
08/01/2024  05:02 PM                67 config.json
08/01/2024  05:02 PM             7,627 initial_conditions.csv
08/01/2024  05:02 PM           160,424 reactions.json
08/01/2024  05:02 PM            33,014 species.json
               4 File(s)        201,132 bytes
               2 Dir(s)  198,050,938,880 bytes free

Use Docker to verify that the JSON files are retrievable, for example:
http://localhost:8000/api/load-example?example=TS1

### CODE REVIEW

**Review by K20shores — CHANGES_REQUESTED** [review]
This is almost correct. The top level folder needs to have a `my_config.json`, like the other three examples have. You **should** (I hope) be able to run your command line music box client on any of the examples by pointing them to the top level folder of any of the examples

File: interactive/api/static/examples/CHAPMAN/camp_data/species.json [hunk 1]
```diff
@@ -3,8 +3,7 @@
     {
       "name": "M",
       "type": "CHEM_SPEC",
-      "tracer type": "CONSTANT",
-      "description": "Third-body molecule. This is any molecule present in the system."
+      "tracer type": "CONSTANT"
```
Comment (K20shores) [inline interactive/api/static/examples/CHAPMAN/camp_data/species.json:6]:
```suggestion
      "tracer type": "THIRD_BODY"
```

It really should be this. If this isn't working then it maybe the wrong version of musica is still being used on your machine

File: interactive/api/static/examples/CHAPMAN/camp_data/config.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "camp-files" : [
-    "camp_data/tolerance.json",
-    "camp_data/species.json",
-    "camp_data/reactions.json"
+    "species.json",
+	"reactions.json",
+    "tolerance.json"
```
Comment (K20shores) [inline interactive/api/static/examples/CHAPMAN/camp_data/config.json:5]: The order shouldn't matter. If it does, we need to correct whatever code thinks the order matters

**Review by K20shores — CHANGES_REQUESTED** [review]

**Review by K20shores — APPROVED** [review]

File: interactive/api/static/examples/FULL_GAS_PHASE/camp_data/config.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "camp-files" : [
-    "camp_data/tolerance.json",
     "camp_data/species.json",
-    "camp_data/reactions.json"
+    "camp_data/reactions.json",
+	"camp_data/tolerance.json"
```
Comment (boulderdaze) [inline interactive/api/static/examples/FULL_GAS_PHASE/camp_data/config.json:5]: A minor thing, but can we use the uniform indentation?

File: interactive/api/static/examples/FLOW_TUBE/camp_data/config.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "camp-files" : [
-    "camp_data/tolerance.json",
     "camp_data/species.json",
-    "camp_data/reactions.json"
+    "camp_data/reactions.json",
+	"camp_data/tolerance.json"
```
Comment (boulderdaze) [inline interactive/api/static/examples/FLOW_TUBE/camp_data/config.json:5]: A minor thing, but can we use the uniform indentation?

File: interactive/api/static/examples/CHAPMAN/camp_data/config.json (hunk 2 above)
Comment (boulderdaze) [inline interactive/api/static/examples/CHAPMAN/camp_data/config.json:5]: A minor thing, but can we use the uniform indentation?

**Review by mattldawson — APPROVED** [review]

**Comment by mattldawson** [comment]
Just noticed the tests are failing - is this something unrelated to this PR?

File: interactive/api/static/examples/CHAPMAN/camp_data/config.json (hunk 2 above)
Comment (carl-drews) [reply interactive/api/static/examples/CHAPMAN/camp_data/config.json:5]: We talked about the order on Slack. There is python code that requires a certain order, and we have plans to fix that. For now, I have put the files in the expected order and this sequence will be forward-compatible.

File: interactive/api/static/examples/FULL_GAS_PHASE/camp_data/config.json (hunk 3 above)
Comment (carl-drews) [reply interactive/api/static/examples/FULL_GAS_PHASE/camp_data/config.json:5]: I'll try tabs in Notepad++.

File: interactive/api/static/examples/CHAPMAN/camp_data/species.json (hunk 1 above)
Comment (carl-drews) [reply interactive/api/static/examples/CHAPMAN/camp_data/species.json:6]: THIRD_BODY now works with musica 0.7.3, and CHAPMAN runs to completion.

File: interactive/api/static/examples/FLOW_TUBE/camp_data/config.json (hunk 4 above)
Comment (carl-drews) [reply interactive/api/static/examples/FLOW_TUBE/camp_data/config.json:5]: Using tabs in Notepad++.

**Review by boulderdaze — APPROVED** [review]

File: interactive/api/static/examples/CHAPMAN/camp_data/config.json (hunk 2 above)
Comment (carl-drews) [reply interactive/api/static/examples/CHAPMAN/camp_data/config.json:5]: Changed to tabs in Notepad++.

**Comment by carl-drews** [comment]
Ready for another round of reviews after addressing requests. Note that TS1 will not run without this python change in music_box.py on about line 721. The addresses a gap in the ordered sequencing:
```

        rate_constants = {}
        for rate in curr_conditions.reaction_rates:

            if (rate.reaction.reaction_type == "PHOTOLYSIS"):
                key = "PHOTO." + rate.reaction.name
            elif (rate.reaction.reaction_type == "LOSS"):
                key = "LOSS." + rate.reaction.name
            elif (rate.reaction.reaction_type == "EMISSION"):
                key = "EMIS." + rate.reaction.name
            rate_constants[key] = rate.rate

        **ordered_rate_constants = (len(rate_constants.keys()) + 1) * [0.0]   # bogus + 1**

```

File: interactive/api/static/examples/CHAPMAN/camp_data/species.json [hunk 5]
```diff
@@ -3,8 +3,8 @@
     {
       "name": "M",
       "type": "CHEM_SPEC",
-      "tracer type": "CONSTANT",
-      "description": "Third-body molecule. This is any molecule present in the system."
+      "tracer type": "THIRD_BODY",
+	  "__description": "Third-body molecule. This is any molecule present in the system."
```
Comment (boulderdaze) [inline interactive/api/static/examples/CHAPMAN/camp_data/species.json:7]: Sorry, I don't want to be obsessed with the formatting but can we use the uniform indentation? If your tab key has the default spacing that is larger than the given format, I think you could use space to match the same indentation

Comment (carl-drews) [reply interactive/api/static/examples/CHAPMAN/camp_data/species.json:7]: How many spaces is the default spacing?

Comment (boulderdaze) [reply interactive/api/static/examples/CHAPMAN/camp_data/species.json:7]: Looks like 2

**Review by K20shores — APPROVED** [review]

**Review by boulderdaze — APPROVED** [review]
Looks good! Thank you for addressing them!

**Comment by carl-drews** [comment]
Formatted with indent = 2 spaces.

**Comment by carl-drews** [comment]
TS1 is ready for further development (restore initial conditions).

**Pull request merged**